File: CellWorld/Models/ComplexConditionModel.cs
namespace CellWorld.Models
{
    public class ComplexConditionModel
    {
        public ConditionModel RightCondition { get; set; }
        public ConditionModel LeftCondition { get; set; }
        public string Operator { get; set; }
        public ComplexConditionModel(){}
    }
}

File: CellWorld/client/src/app/rules/moore-rule/condition-models/complex-condition-model.ts
import { ConditionModel } from "./../condition-model";
export class ComplexConditionModel {
  public LeftCondition: ConditionModel;
  public RightCondition: ConditionModel;
  public Operator: string;
  constructor();
  constructor(
    leftCondition: ConditionModel,
    rightCondition: ConditionModel,
    operator: string
  );
  constructor(
    leftCondition?: ConditionModel,
    rightCondition?: ConditionModel,
    operator?: string
  ) {
    this.LeftCondition = leftCondition || null;
    this.RightCondition = rightCondition || null;
    this.Operator = operator || "and";
  }
}
